cmd/appgw-ingress/main.go

```diff
@@ -161,14 +162,20 @@ func main() {
 
 	if err = azure.WaitForAzureAuth(azClient, maxAuthRetryCount, retryPause); err != nil {
 		if err == azure.ErrAppGatewayNotFound && env.EnableDeployAppGateway {
-			if env.AppGwSubnetID != "" {
-				err = azClient.DeployGatewayWithSubnet(env.AppGwSubnetID)
-			} else if azContext != nil {
-				err = azClient.DeployGatewayWithVnet(azure.ResourceGroup(azContext.VNetResourceGroup), azure.ResourceName(azContext.VNetName), azure.ResourceName(env.AppGwSubnetName), env.AppGwSubnetPrefix)
-			}
+			// create application gateway with retries
+			err = utils.Retry(maxAuthRetryCount, retryPause,
+				func() (utils.Retriable, error) {
+					if env.AppGwSubnetID != "" {
```

Comment:
Can this be a separate function if its okay. ? Might be cleaner to understand the code here